swarm/src/behaviour.rs

```diff
@@ -218,18 +221,26 @@ pub enum NetworkBehaviourAction<TInEvent, TOutEvent> {
         peer_id: PeerId,
     },
 
-    /// Instructs the `Swarm` to send a message to the handler dedicated to the connection with the peer.
+    /// Instructs the `Swarm` to send a message to the handler dedicated to a
+    /// connection with a peer.
     ///
-    /// If the `Swarm` is connected to the peer, the message is delivered to the remote's
-    /// protocol handler. If there is no connection to the peer, the message is ignored.
-    /// To ensure delivery, the `NetworkBehaviour` must keep track of connected peers.
+    /// If the `Swarm` is connected to the peer, the message is delivered to the
+    /// `ProtocolsHandler` instance identified by the peer ID and connection ID.
+    ///
+    /// If there is no connection to the peer, the event is silently dropped.
+    /// To ensure delivery, a `NetworkBehaviour` must keep track of connected peers.
     ///
     /// Note that even if the peer is currently connected, connections can get closed
-    /// at any time and thus the message may not reach its destination.
-    SendEvent {
-        /// The peer to which to send the message.
+    /// at any time and thus the event may not reach a handler.
+    NotifyHandler {
+        /// The peer for whom a `ProtocolsHandler` should be notified.
         peer_id: PeerId,
-        /// The message to send.
+        /// The ID of the connection whose `ProtocolsHandler` to notify.
+        ///
+        /// If `None` and there exist multiple connections to the peer,
+        /// an unspecified choice is made.
+        connection: Option<ConnectionId>,
```

Comment:
I'm ok with the idea of leaving the choice to the swarm here, but I'd suggest splitting this into two variants `NotifyAnyHandler` and `NotifyHandler`.
Alternatively, replacing the `Option` with an enum would be fine as well, but to me it seems more annoying to use from an end-user perspective.

Reply:
Sounds ok to me, I would go with the additional `NotifyAnyHandler` constructor then. It is certainly a bit clearer than seeing `connection: None`.